### PR TITLE
tests: lib: migrating lib tests to new ZTEST API

### PR DESCRIPTION
**Pull request by yerabolu**
Migrating following tests to new ZTEST API.

- sysutil
- sprintf

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

### CODE REVIEW

**Comment by 1xlibiao** [comment]
> Migrating following tests to new ZTEST API.
> 
> * sysutil
> 
> Signed-off-by: Spoorthy Priya Yerabolu [spoorthy.priya.yerabolu@intel.com](mailto:spoorthy.priya.yerabolu@intel.com)

ok,I'll do as soon

File: tests/lib/sprintf/testcase.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ tests:
     integration_platforms:
     - qemu_x86
     platform_exclude: native_posix native_posix_64 nrf52_bsim
-    tags: libc ignore_faults
+    tags: libc ignore_fault
```
Comment (enjiamai) [inline tests/lib/sprintf/testcase.yaml:8]: I think it should be "ignore_faults".

File: tests/lib/sprintf/src/main.c [hunk 2]
```diff
@@ -935,17 +937,18 @@ void test_fwrite(void)
...
  *
  * @details When CONFIG_STDOUT_CONSOLE=n the default
  * stdout hook function _stdout_hook_default() returns EOF.
  */
-void test_EOF(void)
+#else
+ZTEST(sprintf, test_EOF)
 {
 	int ret;
-
```
Comment (enjiamai) [inline tests/lib/sprintf/src/main.c:948]: If this line removed you might meet a compliance check error. Keep it pls.

File: tests/lib/sprintf/src/main.c [hunk 3]
```diff
@@ -785,25 +803,9 @@ void test_fprintf(void)
 }
 
 
-/**
- *
- * @brief Test vfprintf function
- *
- */
```
Comment (enjiamai) [inline tests/lib/sprintf/src/main.c:792]: Line 788 ~ 792 should be kept, right?

File: tests/lib/sprintf/prj.conf [hunk 4]
```diff
@@ -1,4 +1,6 @@
 CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y
 CONFIG_FPU=y
 CONFIG_TEST_USERSPACE=y
 CONFIG_ZTEST_FATAL_HOOK=y
+CONFIG_STDOUT_CONSOLE=y
```
Comment (enjiamai) [inline tests/lib/sprintf/prj.conf:6]: Do we need to add this? It seems like being a little bit different with the original test.

**Review by enjiamai — CHANGES_REQUESTED** [review]
LGTM,  just some minor problem, thanks!

File: tests/lib/sprintf/prj.conf (hunk 4 above)
Comment (yerabolu) [reply tests/lib/sprintf/prj.conf]: Yes, i was debugging something else and forgot to remove this before pushing the code. Removed it now.

File: tests/lib/sprintf/src/main.c (hunk 2 above)
Comment (yerabolu) [reply tests/lib/sprintf/src/main.c:948]: Thank You Enjia for reviewing it!! Pushed all the suggested changes.

File: tests/lib/sprintf/prj.conf [hunk 5]
```diff
@@ -1,4 +1,6 @@
 CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y
 CONFIG_FPU=y
 CONFIG_TEST_USERSPACE=y
 CONFIG_ZTEST_FATAL_HOOK=y
+#CONFIG_STDOUT_CONSOLE=y
```
Comment (enjiamai) [inline tests/lib/sprintf/prj.conf:6]: I think remove it is a better choice, thanks.

**Review by enjiamai — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Comment by enjiamai** [comment]
Hi @yerabolu , could you please rebase it? Thanks.

**Comment by enjiamai** [comment]
Hi, @yerabolu , need rebase?

**Comment by yerabolu** [comment]
Sorry, i am having issues with rebase and changes get committed as another commit. Will close this PR and open a new one.

**Pull request closed without merging**